plugin.rb

```diff
@@ -126,18 +156,52 @@ def command
-            render json: { text: "*#{DiscourseSlack::Slack.filter_to_past(cmd).capitalize} all categories* on this channel." }
-          elsif (category && guardian.can_see_category?(category))
-            DiscourseSlack::Slack.set_filter_by_id(category.id, channel, cmd, params[:channel_id])
-            render json: { text: "*#{DiscourseSlack::Slack.filter_to_past(cmd).capitalize}* category *#{category.name}*" }
+          name = tokens[1]
+          if name.start_with?('tag:')
+            tag_name = name.sub! 'tag:', ''
+            tag = Tag.find_by({name: tag_name})
+            if tag
+              row = ::PluginStoreRow.find_by({plugin_name: PLUGIN_NAME, key: "_tag_#{tag.name}_#{channel}"})
+              if row
+                data = get_filter(row.value)
+                unless data['filter'] === cmd
+                  data['tags'] = data['tags'].delete(tag.name)
+                  ::PluginStore.set(PLUGIN_NAME, "filter_#{row.value}", data)
+                  ::PluginStore.remove(PLUGIN_NAME, "_tag_#{tag.name}_#{channel}")
+                end
+              end
+              DiscourseSlack::Slack.set_filter(channel, cmd, nil, [tag.name])
+              render json: { text: "*#{DiscourseSlack::Slack.filter_to_past(cmd).capitalize}* tag *#{tag.name}*" }
+            else
+              render json: { text: "I can't find the *#{tag_name}* tag." }
+            end
           else
-            # TODO DRY (easy)
-            cat_list = (CategoryList.new(Guardian.new User.find_by_username(SiteSetting.slack_discourse_username)).categories.map { |c| c.slug }).join(', ')
-            render json: { text: "I can't find the *#{tokens[1]}* category. Did you mean: #{cat_list}" }
+            cat_name = name
+            category = Category.find_by({slug: cat_name})
+            category_id = -1
+            category_id = 0 if (cat_name.casecmp("all") === 0)
+            category_id = category.id if (category && guardian.can_see_category?(category))
+            if category_id > -1
+              row = ::PluginStoreRow.find_by({plugin_name: PLUGIN_NAME, key: "_category_#{category_id}_#{channel}"})
+              if row
+                data = get_filter(row.value)
+                unless data['filter'] === cmd
+                  data['category_id'] = nil
+                  ::PluginStore.set(PLUGIN_NAME, "filter_#{row.value}", data)
+                  ::PluginStore.remove(PLUGIN_NAME, "_category_#{category_id}_#{channel}")
+                end
+              end
+              DiscourseSlack::Slack.set_filter(channel, cmd, category_id)
+              if category_id == 0
+                render json: { text: "*#{DiscourseSlack::Slack.filter_to_past(cmd).capitalize} all categories* on this channel." }
+              else
+                render json: { text: "*#{DiscourseSlack::Slack.filter_to_past(cmd).capitalize}* category *#{category.name}*" }
+              end
+            else
+              # TODO DRY (easy)
```

Comment:
I almost done everything except it. I can't understand what exactly I have to do here. Can you @tgxworld please give me more details?